### PR TITLE
:sparkles: [Feature] Add requested feature with transformation for both value and error

### PR DESCRIPTION
**Pull request by kittinunf**
### What's in this PR?

This PR adds feature request for `mapBoth` to be able to transform the type for both sides of the Type.

### CODE REVIEW

File: result/src/commonTest/kotlin/com/github/kittinunf/result/ResultTest.kt [hunk 1]
```diff
@@ -269,6 +269,27 @@ class ResultTest {
...
+        assertEquals(11, transformedSuccess.get())
+        assertEquals(null, transformedSuccess.getFailureOrNull())
+
+        val f = Result.of<Int, Throwable> { 1 / 0 } //this should have arithmetic exception
+
+        val transformedFailure = f.mapBoth({ it > 1}, { RuntimeException("changed!") })
+
+        assertIs<ArithmeticException>(f.getFailureOrNull())
+        assertEquals(null, transformedFailure.getOrNull())
+        assertEquals("changed!", transformedFailure.getFailureOrNull()!!.message)
```
Comment (lopes-felipe) [inline result/src/commonTest/kotlin/com/github/kittinunf/result/ResultTest.kt:290]: What about to assert the `transformedFailure`'s type as well?

**Review by lopes-felipe — APPROVED** [review]
Nice feature!

**Comment by iNoles** [comment]
JS doesn't like `` on function with whitespace

File: result/src/commonTest/kotlin/com/github/kittinunf/result/ResultTest.kt (hunk 1 above)
Comment (kittinunf) [reply result/src/commonTest/kotlin/com/github/kittinunf/result/ResultTest.kt:290]: True, I will fix this 👍

Comment (kittinunf) [reply result/src/commonTest/kotlin/com/github/kittinunf/result/ResultTest.kt]: Fixed! 5650e4e

**Review by lopes-felipe — APPROVED** [review]

**Comment by kittinunf** [comment]
Thanks! @lopes-felipe

**Pull request merged**